### ISSUE TITLE
Error while changing the cover image on Product edit page

### ISSUE THREAD

**Issue by HyperBeats** [PrestaShop/PrestaShop#34534]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hello

I currently have an error on my site when I want to change the cover image of my items I also specify that this store and a store that was prestashop 1.7 that I migrated to prestashop 8.1.1

Here is the error.
![image](https://github.com/PrestaShop/PrestaShop/assets/45314346/3a64346b-f824-4f1c-8908-1945c9830de4)

https://github.com/PrestaShop/PrestaShop/assets/45314346/87b216bc-cb3a-479a-9d62-a75471e23cac

### Steps to reproduce

1 . allez dans l'onglet produit
2 . modifier un produit
3 . upload des image pour le produits
4 . selectionné l'image de cover

### PrestaShop version(s) where the bug happened

8.1.1

### PHP version(s) where the bug happened

8.0

### Your company or customer's name goes here (if applicable).

ApsaraFrance

**Comment by florine2623** [comment]
Hello @HyperBeats ,

I can't reproduce your issue.

https://github.com/PrestaShop/PrestaShop/assets/16019289/d606cc54-5015-4e9c-b4e1-7fc226c5430c




I see that you have installed an external module on this page. You can discuss with the developer of the module. The issue might be related to it.

We use GitHub issues only to discuss about bugs and new features in the PrestaShop project. If you need support, you can find resources to help you on this page: [Get help with PrestaShop](https://www.prestashop-project.org/support/).

Alternatively, you can also ask for help in the [community forums](https://www.prestashop.com/forums/) or in the [public Slack channel](https://github.com/PrestaShop/open-source/tree/master/slack).

Thank you

**Comment by jix31** [comment]
I have exactly the same bug with a fresh installation

when i want to set cover product image / or save caption text (the text is saved but i he a red error message) / and move product image don't works ( i can't change the order of the image after upload the  products)

i have this red message : function(){return r&&o(s,r),(e.migrateIsPatchEnabled(s)?i:a||e.noop).apply(this,arguments)}
<img width="712" alt="Capture d’écran 2023-11-21 à 18 27 01" src="https://github.com/PrestaShop/PrestaShop/assets/151663961/4ec4ed3f-3d16-40e5-a871-fba0755d78b5">

PrestaShop version(s) where the bug happened
8.1.2

PHP version(s) where the bug happened
8.1.24

**Comment by TamasFelix** [comment]
I encountered a similar issue on the website [www.fashionland.ro](url)
function(){return r&&o(s,r),(e.migrateIsPatchEnabled(s)?i:a||e.noop).apply(this,arguments)}
The problem occurred when I edited or added another image to the representative image. I re-uploaded the images to re-establish the connection and activated the old version of product editing.

One of the major issues is that the modified or deleted images in the database (DB) persist in the table.

The problem originates from the new core version; the links are not properly established.
